lib/opensearch-aws-sigv4.rb

```diff
@@ -17,6 +17,74 @@
 
 module OpenSearch
   module Aws
+    module Transport
+      # Extending OpenSearch::Transport::Client to sign requests with SigV4.
+      class Sigv4Client < ::OpenSearch::Transport::Client
+        attr_accessor :sigv4_signer
+
+        def initialize(transport_args, sigv4_signer, sigv4_debug, &block)
```

Comment:
It's really uncommon to have a constructor like this. It is difficult to evolve, and does not match the superclass.

Next, the `sigv4_debug` parameter is problematic because it directs security-related information to logger in an unconventional way. Get rid of it and always `log` Sigv4 information at debug level. This way when auth fails using this client the user can turn on debug level logging and narrow things down without modifying the code.

So, the constructor should match the one of `OpenSearch::Transport::Client`

```ruby
def initialize(arguments = {}, &block)
   arguments = arguments.dup
   signer = arguments.delete(:signer) 
   raise ... unless signer
   super arguments, block
end
```